modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs.js

```diff
@@ -12,769 +12,616 @@
-					}
-				},
+		return automaticURLInput.value === 'true';
+	}
 
-				_configureAbstract(event) {
-					var instance = this;
+	_beforePublishBtnClick() {
+		const form = Liferay.Form.get(`${this._config.namespace}fm`);
 
-					var target = event.target;
+		form.addRule(
+			`${this._config.namespace}title`,
+			'required',
+			this._config.strings.titleRequiredAtPublish
+		);
+	}
 
-					var description = instance._customDescription;
+	_beforeSaveBtnClick() {
+		const form = Liferay.Form.get(`${this._config.namespace}fm`);
 
-					instance._shortenDescription = target.val() === 'false';
+		form.removeRule(`${this._config.namespace}title`, 'required');
+	}
 
-					if (instance._shortenDescription) {
-						instance._customDescription = instance
-							.one('#description')
-							.val();
+	_bindUI() {
+		const events = this._events;
+
+		this._captionNode = this._rootNode.querySelector(
+			'.cover-image-caption'
+		);
+
+		Liferay.on('coverImageDeleted', this._removeCaption, this);
+		Liferay.on(
+			['coverImageUploaded', 'coverImageSelected'],
+			this._showCaption,
+			this
+		);
+
+		const publishButton = this._getElementById('publishButton');
+
+		if (publishButton) {
+			events.add(publishButton, STR_CLICK, () => {
+				this._beforePublishBtnClick();
+				this._checkImagesBeforeSave(false, false);
+			});
+		}
+
+		const saveButton = this._getElementById('saveButton');
+
+		if (saveButton) {
+			events.add(saveButton, STR_CLICK, () => {
+				this._beforeSaveBtnClick();
+				this._checkImagesBeforeSave(true, false);
+			});
+		}
+
+		const customAbstractOptions = document.querySelectorAll(
+			`input[name=${this._config.namespace}customAbstract]`
+		);
+
+		if (customAbstractOptions.length) {
+			customAbstractOptions.forEach((option) => {
+				events.add(
+					option,
+					STR_CHANGE,
+					this._configureAbstract.bind(this)
+				);
+			});
+		}
+
+		const urlOptions = document.querySelectorAll(
+			`#${this._config.namespace}urlOptions input`
+		);
+
+		if (urlOptions.length) {
+			urlOptions.forEach((option) => {
+				events.add(
+					option,
+					STR_CHANGE,
+					this._onChangeURLOptions.bind(this)
+				);
+			});
+		}
+
+		const titleInput = this._getElementById('title');
+
+		if (titleInput) {
+			events.add(titleInput, STR_CHANGE, (event) => {
+				this.updateFriendlyURL(event.target.value);
+			});
+		}
+
+		const descriptionInput = this._getElementById('description');
+
+		if (descriptionInput) {
+			events.add(descriptionInput, STR_CHANGE, (event) => {
+				this.setCustomDescription(event.target.value);
+			});
+		}
+	}
 
-						description = window[
-							instance.ns('contentEditor')
-						].getText();
-					}
+	_checkImagesBeforeSave(draft, ajax) {
+		const instance = this;
 
-					instance.setDescription(description);
-				},
+		const tempImages = this._getTempImages();
 
-				_getContentImages(content) {
-					var contentDom = document.createElement('div');
+		if (tempImages.length) {
```

Comment:
Not blocking, but maybe this could use `_hasTempImages` or change the single usage of `hasTempImages` with `_getTempImages().length`. I would go with this in both places, but up to you!